test/sparse/higherorderfns.jl

```diff
@@ -187,6 +188,10 @@ end
             broadcast!(+, Z, X, Y); Z = sparse(fZ) # warmup for @allocated
             @test (@allocated broadcast!(+, Z, X, Y)) == 0
             @test broadcast!(+, Z, X, Y) == sparse(broadcast!(+, fZ, fX, fY))
+            broadcast!(-, fZ, fX, fY); Z = sparse(fZ)
+            broadcast!(-, Z, X, Y); Z = sparse(fZ) # warmup for @allocated
+            @test (@allocated broadcast!(-, Z, X, Y)) == 0
+            @test broadcast!(-, Z, X, Y) == sparse(broadcast!(-, fZ, fX, fY))
```

Comment:
These tests broadcasting `-` exercise the same code paths as those for `+`; together `+` and `*` cover the full set of code paths. Perhaps nix these?

Reply:
[This](https://codecov.io/gh/JuliaLang/julia/src/55ec9fea4b25d785ab7bf4f0a9c23676c570eb18/base/sparse/higherorderfns.jl#L175) is what I was trying to cover here

Reply:
To spot test that line, try
```julia
a, b = sprand(5, 0.5),  sprand(5, 0.5)
@test broadcast(-, a, b) == broadcast(-, Vector(a), Vector(b))
```
The tests above are quite heavy and indeed exercise the same code paths as the `+` tests. Best!

Reply:
Cheers! Thanks for the explanation.

Reply:
actually, wait a second ... why isn't line 176 in this file catching this path?

Reply:
How does one see this pull request's measured impact on coverage? The `broadcast(-, X, Y)` test you added seems like it should do the trick, as it should generate the following call
```julia
julia> @which broadcast(-, SparseVector{Float32,Int32}(sprand(1, 1.0)), sprand(1, 1.0))
broadcast(::typeof(-), x::SparseVector, y::SparseVector) in Base.SparseArrays.HigherOrderFns at sparse/higherorderfns.jl:175
```
which should cover the line you would like to cover.

In any case, I would remove the `broadcast(-, X, Y)` addition in favor of the spot test above, as the `broadcast(-, X, Y)` line is quite heavy and primarily exercises the same code as the adjacent `broadcast(+, X, Y)`. Best!